### DISCUSSION TITLE
How can I run it on Macbook (m1 air)?

### DISCUSSION THREAD

**Discussion by tdemir97** [lllyasviel/stable-diffusion-webui-forge#324] [Q&A]
Any tutorial for a newbie? :(

**Comment by ophoria** [comment]
Havent tried it again on my M1 recently, but last time I did use SD on it wasn't really a nice experience - took forever compared with an actual graphicscard on my desktop.

**Comment by MonetEinsley** [comment]
I got it working on my MacBook Air M1 by editing the webui-user.sh file at around line 12:
# Commandline arguments for webui.py, for example: export COMMANDLINE_ARGS="--medvram --opt-split-attention"
A1111_Home=/Users/moneteinsley/stable-diffusion-webui
export COMMANDLINE_ARGS=" --skip-torch-cuda-test 
--ckpt-dir $A1111_Home/models/Stable-diffusion 
--hypernetwork-dir $A1111_Home/models/hypernetworks 
--embeddings-dir $A1111_Home/embeddings 
--lora-dir $A1111_Home/models/Lora 
--vae-dir $A1111_Home/models/VAE"
______________

I’m using it as a separate thing from my Automatic 1111 installation, hence why I have these directory pointers here. It works for me using SD1.5, but I’m still having trouble using SDXL even on very low resolutions. It gives me a TypeError: ‘NoneType’ object is not iterable

That being said, I see no difference in terms of performance compared to my A1111 installation where I have the arguments:
export COMMANDLINE_ARGS="--skip-torch-cuda-test --upcast-sampling --opt-sub-quad-attention --use-cpu interrogate --medvram-sdxl --no-half-vae"
And unlike my current installation of Forge, SDXL does barely work on my A1111 installation at 512x512.
Right now, using an SD1.5 LCM checkpoint, it takes about 10-16 minutes to generate 1 image both on Forge and A1111. (This is with 8-16 steps for the first pass at 512x512, 8 steps for hires fix to 1024x1024, and 8 steps for any ADetailer face fixes at 512x512)

I know it said in the README that the CMD flags pertaining to resource management (medvram, lowvram, etc..) were all removed, but I think I want to try using some of the other arguments like --opt-sub-quad-attention into the Forge webui-user.sh file, to see if there’s any improvement. If I can get SDXL working on it, that’ll be good enough for me.
_________
Edit: Thanks to dermesut for pointing this out, but you can further simplify the webui-user.sh file to something like this:

A1111_Home=/Users/moneteinsley/stable-diffusion-webui
export COMMANDLINE_ARGS=" --skip-torch-cuda-test
--forge-ref-a1111-home $A1111_Home
--gfpgan-model $A1111_Home/models/GFPGAN
"
And that will basically allow you to use your models/controlnets, etc from your A1111 folder for use in Forge
